model/span_test.go

```diff
@@ -0,0 +1,75 @@
+package model
+
+import (
+	"encoding/json"
+	"errors"
+	"net"
+	"reflect"
+	"testing"
+	"time"
+)
+
+func TestJSON(t *testing.T) {
+	var (
+		span1    SpanModel
+		span2    SpanModel
+		parentID = ID(1003)
+		sampled  = true
+		tags     = make(map[string]string)
+	)
+	tags["myKey"] = "myValue"
+	tags["another"] = "tag"
+
+	span1 = SpanModel{
+		SpanContext: SpanContext{
+			TraceID: TraceID{
+				High: 1001,
+				Low:  1002,
+			},
+			ID:       ID(1004),
+			ParentID: &parentID,
+			Debug:    true,
+			Sampled:  &sampled,
+			Err:      errors.New("dummy"),
+		},
+		Name:      "myMethod",
+		Kind:      Server,
+		Timestamp: time.Now().Add(-100 * time.Millisecond),
+		Duration:  50 * time.Millisecond,
+		Shared:    true,
+		LocalEndpoint: &Endpoint{
+			ServiceName: "myService",
+			IPv4:        net.IPv4(127, 0, 0, 1),
```

Comment:
suppose one can still pass literals here? (string literals) or does this enforce types? In java, constructing these objects can be expensive, not sure about go.

Reply:
IPv4 is a byte array and not expensive to populate

Reply:
If we passed a literal we had to validate it somewhere, I think it is better to use this standard library object and the marshalling is not expensive.